src/chol.jl

```diff
@@ -1,6 +1,10 @@
 CholType{T,S<:AbstractMatrix} = Cholesky{T,S}
 chol_lower(a::Matrix) = cholesky(a).L
 
+# always use the stored cholesky factor, not a copy
+@inline chol_lower(a::CholType) = a.uplo === 'L' ? a.L : transpose(a.U)
```

Comment:
```suggestion
chol_lower(a::CholType) = a.uplo === 'L' ? a.L : a.U'
```
I think we should only add annotations if we know they are needed. I'd think the heuristic would handle this one just fine.